### ISSUE TITLE
Detekt when unneccessary `inner` modifier is used for a class

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8896]
## Expected Behavior of the rule
```kt
class A {
    inner class B(private val b: Int)
}
```
here class `B`'s `inner` modifier should not be allowed as `B` is not referencing to any of the `A` variable or property(intellij gives warning for this)

**Comment by atulgpt** [comment]
There is already one rule `UnnecessaryInnerClass`
